### ISSUE TITLE
Delivery callback called with an invalid error when producing without ACL permissions

### ISSUE THREAD

**Issue by mensfeld** [appsignal/rdkafka-ruby#92]
Version: current master

Reproduction:

- clone https://github.com/paneq/kafka_docker_sasl_plain
- run `docker-compose -f test.yml up`
- build rdkafka with sasl support
- execute:

```ruby
config = {
    :'bootstrap.servers' => '127.0.0.1:9092',
    :'request.required.acks' => 1,
    :'sasl.mechanisms' => 'PLAIN',
    :'sasl.username' => 'edward',
    :'sasl.password' => 'edward',
    :'security.protocol' => 'sasl_plaintext',
    :'message.timeout.ms' => 100
  }
producer = Rdkafka::Config.new(config).producer
delivery_handles = []

100.times do |i|
  puts "Producing message #{i}"
  delivery_handles << producer.produce(
      topic:   "ruby-test-topic",
      payload: "Payload #{i}",
      key:     "Key #{i}"
  )
end

delivery_handles.each(&:wait)

Rdkafka::RdkafkaError (Local: Message timed out (msg_timed_out))
```

We never get a proper failure (something like authorization failed). Instead, we /librdkafka wait forever (till the `message.timeout.ms`).

This behavior can be really confusing as one would expect to get a clear error message indicating a lack of permissions.

ref https://github.com/karafka/waterdrop/issues/108

**Comment by thijsc** [comment]
Has this been fixed by #86?

We don't use ACLs so don't have a test setup handy for it myself.

**Comment by thijsc** [comment]
Closing, feel free to reopen if this is still relevant.
